packages/astro/src/core/sync/write-files.ts

```diff
@@ -0,0 +1,78 @@
+import type fsMod from 'node:fs';
+import { dirname, relative } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { bold } from 'kleur/colors';
+import { normalizePath } from 'vite';
+import type { AstroSettings } from '../../@types/astro.js';
+import type { Logger } from '../logger/core.js';
+import { REFERENCE_FILE } from './constants.js';
+import { AstroError, AstroErrorData } from '../errors/index.js';
+
+export async function writeFiles(settings: AstroSettings, fs: typeof fsMod, logger: Logger) {
+	try {
+		writeInjectedTypes(settings, fs);
+		await setUpEnvTs(settings, fs, logger);
+	} catch (e) {
+		throw new AstroError(AstroErrorData.UnknownError, { cause: e });
```

Comment:
@ematipico what error should I throw here?

Reply:
I think a generic error that says that Astro couldn't read or write files to disk should be fine. The `cause` should give us the stacktrace of the source, which should be enough for us. 

You can create a new Astro error if you want.